react/features/authentication/middleware.web.js

```diff
@@ -0,0 +1,137 @@
+// @flow
+
+import { appNavigate } from '../app/actions';
+import {
+    CONFERENCE_FAILED,
+    CONFERENCE_JOINED,
+    CONFERENCE_LEFT
+} from '../base/conference';
+import { CONNECTION_ESTABLISHED } from '../base/connection';
+import { hideDialog, isDialogOpen } from '../base/dialog';
+import {
+    JitsiConferenceErrors
+} from '../base/lib-jitsi-meet';
+import { MiddlewareRegistry } from '../base/redux';
+
+import {
+    CANCEL_LOGIN,
+    STOP_WAIT_FOR_OWNER,
+    WAIT_FOR_OWNER
+} from './actionTypes';
+import {
+    stopWaitForOwner,
+    waitForOwner
+} from './actions.native';
+import {
+    hideLoginDialog,
+    openWaitForOwnerDialog
+} from './actions.web';
+import { LoginDialog, WaitForOwnerDialog } from './components';
+
+/**
+ * Middleware that captures connection or conference failed errors and controls
+ * {@link WaitForOwnerDialog} and {@link LoginDialog}.
+ *
+ * FIXME Some of the complexity was introduced by the lack of dialog stacking.
+ *
+ * @param {Store} store - Redux store.
+ * @returns {Function}
+ */
+MiddlewareRegistry.register(store => next => action => {
+    switch (action.type) {
+
+    case CANCEL_LOGIN: {
+        // Checks if WaitForOwnerDialog is open
+        if (!isDialogOpen(store, WaitForOwnerDialog)) {
+            if (isWaitingForOwner(store)) {
+
+                store.dispatch(openWaitForOwnerDialog());
+
+                return next(action);
+            }
+
+            // Go back to the app's entry point.
+            store.dispatch(hideLoginDialog());
+
+            store.dispatch(appNavigate(undefined));
```

Comment:
Shouldn't this be `maybeRedirectToWelcomePage` ?

Reply:
Changed.